### PR TITLE
add fast-checkpoint examples for nebulaml

### PR DESCRIPTION
**Pull request by alexwong2024**
# Description

In this PR, we add a bert example to demonstrate Nebula fast-checkpointing examples.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: cli/jobs/nebulaml/bert-pretrain/job.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Training job submission via AML CLI v2
+
+$schema: https://azuremlschemas.azureedge.net/latest/commandJob.schema.json
+
+command: pip list && python pretrain_glue.py --save_steps 20 --deepspeed ds_config_bertbase.json --num_train_epochs 100 --output_dir outputs --disable_tqdm 1 --local_rank $RANK --evaluation_strategy "epoch" --logging_strategy "epoch" --per_device_train_batch_size 532 --gradient_accumulation_steps 1 --per_device_eval_batch_size 532 --learning_rate 3e-05 --adam_beta1 0.8 --adam_beta2 0.999 --weight_decay 3e-07 --warmup_steps 500 --fp16 --logging_steps 1000 --model_checkpoint "bert-base-uncased"
+
+experiment_name: bert-pretrain-nebula-ds-optimal
+environment: {your environment} # Should replace your environment
```
Comment (harneetvirk) [inline cli/jobs/nebulaml/bert-pretrain/job.yml:8]: Are you planning to add any scheduled validation of these workflows. 
If yes, how would these values be updated at the runtime with some valid values.
If not, how would be able to make sure the samples are in the valid/working state.

File: cli/jobs/nebulaml/bert-pretrain/src/train.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import json
+
+
+if __name__ == "__main__":
+
+    parser = HfArgumentParser(TrainingArguments)
+    training_args, args = parser.parse_args_into_dataclasses(return_remaining_strings=True)
+
+    #bookcorpus = load_dataset("bookcorpus", split="train")
+    #wikide = load_dataset("wikipedia", "20220301.de", split="train")
```
Comment (harneetvirk) [inline cli/jobs/nebulaml/bert-pretrain/src/train.py:14]: nit: Remove the commented calls if not required.

File: cli/jobs/nebulaml/bert-pretrain/src/environment/env.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  path: ./context/
+tags:
+  os: ubuntu
+  os_version: 20.04
+  hpcx: 2.10
+  mpi: openmpi
+  mpi_version: 4.1.2rc4
+  ucx: 1.12.0
+  cuda: 11.7
+  cublas: 11.10.3.66
```
Comment (harneetvirk) [inline cli/jobs/nebulaml/bert-pretrain/src/environment/env.yml:14]: Please run `python readme.py` from cli directory to determine what all workflows are being added.

File: cli/jobs/nebulaml/bert-pretrain/src/environment/context/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+RUN pip install evaluate
+
+RUN pip install git+https://github.com/FreyaRao/DeepSpeed.git@qrao/fix_save_16mit_model_error
+
+# Install pip dependencies
+RUN pip install 'ipykernel~=6.0' \
+                'azureml-core==1.48.0' \
+				'azureml-dataset-runtime==1.48.0' \
+                'azureml-defaults==1.48.0' \
+				'azure-ml==0.0.1' \
```
[the remark this answers is not in the document]
Comment (alexwong2024) [reply cli/jobs/nebulaml/bert-pretrain/src/environment/context/Dockerfile]: not needed, removed in latest code.

File: cli/jobs/nebulaml/bert-pretrain/src/environment/context/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+RUN pip install accelerate
+RUN pip install scikit-learn
+RUN pip install apache_beam
+RUN pip install evaluate
+
+RUN pip install git+https://github.com/FreyaRao/DeepSpeed.git@qrao/fix_save_16mit_model_error
+
+# Install pip dependencies
+RUN pip install 'ipykernel~=6.0' \
+                'azureml-core==1.48.0' \
```
[the remark this answers is not in the document]
Comment (alexwong2024) [reply cli/jobs/nebulaml/bert-pretrain/src/environment/context/Dockerfile]: strict pinning removed.

File: cli/jobs/nebulaml/bert-pretrain/src/environment/context/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+				'azure-ml-component==0.9.16.post2' \
+                'azureml-mlflow==1.48.0' \
+		'azureml-contrib-services==1.48.0' \
+		        'azureml-contrib-services==1.48.0' \
+                'torch-tb-profiler~=0.4.0' \
+				'py-spy==0.3.12' \
+		        'debugpy~=1.6.3'
+
+RUN pip install \
+	azure-ai-ml==1.2.0 \
```
[the remark this answers is not in the document]
Comment (alexwong2024) [reply cli/jobs/nebulaml/bert-pretrain/src/environment/context/Dockerfile]: remove strict pinning, when install should use the latest version.

File: cli/jobs/nebulaml/bert-pretrain/src/train.py (hunk 2 above)
Comment (alexwong2024) [reply cli/jobs/nebulaml/bert-pretrain/src/train.py]: removed.

File: cli/jobs/nebulaml/bert-pretrain/job.yml (hunk 1 above)
Comment (alexwong2024) [reply cli/jobs/nebulaml/bert-pretrain/job.yml]: users can check out the logs to validate, the guidance is in readme.

File: cli/jobs/nebulaml/bert-pretrain/src/environment/env.yml (hunk 3 above)
Comment (alexwong2024) [reply cli/jobs/nebulaml/bert-pretrain/src/environment/env.yml:14]: generated with readme.py

**Comment by ZikeiWong** [comment]
Hi @harneetvirk, could you please review this PR so that we can show our customers the samples to get started? Thanks

**Review by ZikeiWong — APPROVED** [review]

**Review by cloga — APPROVED** [review]

**Pull request merged**